### DISCUSSION TITLE
distortion simulation in qucs-s

### DISCUSSION THREAD

**Discussion by bnilsson11** [ra3xdh/qucs_s#787] [Q&A] [answered]
I am trying to use the DISTORTION Simulation in qucs-s.
The qucs-s manual (https://qucs-s-help.readthedocs.io/en/latest/ASim.html) says I cannot use the regular AC source for this, I must use a special SPICE source.
This source is to be found in the "Spice components" group.
My problem is that I cannot find any such component group.
My qucs-s Components contain the following:

- Lumped components
- Sources
- Probes
- RF components
- Transmission Lines
- Nonlinear components
- Microelectronics
- Digital components
- File components
- Simulations
- Equations
- XSPICE devices
- SPICE netlist sections
- Paintings

Is this special AC source placed in a different Component group than described in the manual?  Or is my qucs-s installation incomplete?

I am using qucs-s-24.2.1 built for macOS 14.5 (23F79), and ngspice 42.

BN

**Comment by ra3xdh** [comment, marked as answer]
The guide at the readthedocs.io is outdated. You have to use the red one *V source* device. Enter SPICE string as shown in the example in its properties. You can also find up-to-date DDISTO simulation example in the `share/examples/ngspice/Distortion.sch` in the Qucs-S distribution root directory. 

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/50e997a8-b602-4744-aa84-bf512e52ac9c)

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/2763c309-59ab-4029-9214-c3747d55188b)

**Comment by bnilsson11** [comment]
Thank you for the clarification.
Is there any descriptive text regarding the Distortion.sch example?
I have only just started touching this subject, and I would like to understand what is going on.
A link to a non-outdated manual/help would be greatly appreciated.

**Comment by ra3xdh** [comment]
The new reference guide for Qucs-S is under construction #554 and not finished yet. The distortion simulation input and output variables in the example above follow the Ngspice manual. You may look at the Ngspice manual chapter 15.3.3 for explanation https://ngspice.sourceforge.io/docs/ngspice-html-manual/manual.xhtml

You may also find useful the following resources about Qucs-S:

* Qucs-S tutorial https://ra3xdh.github.io/pdf/qucs_s_tutorial.pdf
* Video tutorial from Kasper Nielsen: https://www.youtube.com/watch?v=90RaVy38DB8
* Advanced simulation guide: https://habr.com/ru/articles/789452/ Sorry, only Russian version is available.

**Comment by bnilsson11** [comment]
Thanks, very useful indeed.


> 25 juni 2024 kl. 08:31 skrev Vadim Kuznetsov ***@***.***>:
> 
> 
> The new reference guide for Qucs-S is under construction #554 <https://github.com/ra3xdh/qucs_s/issues/554> and not finished yet. The distortion simulation input and output variables in the example above follow the Ngspice manual. You may look at the Ngspice manual chapter 15.3.3 for explanation https://ngspice.sourceforge.io/docs/ngspice-html-manual/manual.xhtml
> 
> You may also find useful the following resources about Qucs-S:
> 
> Qucs-S tutorial https://ra3xdh.github.io/pdf/qucs_s_tutorial.pdf
> Video tutorial from Kasper Nielsen: https://www.youtube.com/watch?v=90RaVy38DB8
> Advanced simulation guide: https://habr.com/ru/articles/789452/ Sorry, only Russian version is available.
> —
> Reply to this email directly, view it on GitHub <https://github.com/ra3xdh/qucs_s/discussions/787#discussioncomment-9867642>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAZXLDRPITKEOOCLFD5YAV3ZJEFF3AVCNFSM6AAAAABJ2HPVZ6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TQNRXGY2DE>.
> You are receiving this because you authored the thread.
> 

Bengt Nilsson
Ansåsvägen 8
43853 Hindås
Sweden
***@***.***
